AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/KeyManagerTests.java

```diff
@@ -0,0 +1,201 @@
+        assertNotNull("byte[] of DPK should not be null", dpk.getKey());
+        assertTrue("byte[] of DPK should not be empty", dpk.getKey().length > 0);
+
+        boolean keyExists = manager.keyExists();
+        assertTrue("DPK should exist but does not exist", keyExists);
+    }
+
+    public void testDPKClear() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+
+        String password = "passw0rd";
+
+        EncryptionKey dpk = manager.generateAndSaveKeyProtectedByPassword(password);
+        assertNotNull("DPK should not be null", dpk);
+        assertNotNull("byte[] of DPK should not be null", dpk.getKey());
+        assertTrue("byte[] of DPK should not be empty", dpk.getKey().length > 0);
+
+        boolean keyExists = manager.keyExists();
+        assertTrue("DPK should exist but does not exist", keyExists);
+
+        EncryptionKey storedDPK = manager.loadKeyUsingPassword(password);
+        assertNotNull("DPK should not be null", storedDPK);
+        assertNotNull("byte[] of DPK should not be null", storedDPK.getKey());
+        assertTrue("byte[] of DPK should not be empty", storedDPK.getKey().length > 0);
+
+        boolean clearSuccess = manager.clearKey();
+        assertTrue("clear key failed", clearSuccess);
+
+        storedDPK = manager.loadKeyUsingPassword(password);
+        assertNull("DPK should be null", storedDPK);
+
+        keyExists = manager.keyExists();
+        assertFalse("DPK should not exist but does not exist", keyExists);
+    }
+
+    public void testDPKRecreateAfterClear() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+
+        String password = "passw0rd";
+
+        EncryptionKey dpk = manager.generateAndSaveKeyProtectedByPassword(password);
+        assertNotNull("DPK should not be null", dpk);
+        assertNotNull("byte[] of DPK should not be null", dpk.getKey());
+        assertTrue("byte[] of DPK should not be empty", dpk.getKey().length > 0);
+
+        boolean keyExists = manager.keyExists();
+        assertTrue("DPK should exist but does not exist", keyExists);
+
+        boolean clearSuccess = manager.clearKey();
+        assertTrue("clear key failed", clearSuccess);
+
+        keyExists = manager.keyExists();
+        assertFalse("DPK should not exist but does not exist", keyExists);
+
+        EncryptionKey newDPK = manager.generateAndSaveKeyProtectedByPassword(password);
+        assertNotNull("DPK should not be null", newDPK);
+        assertNotNull("byte[] of DPK should not be null", newDPK.getKey());
+        assertTrue("byte[] of DPK should not be empty", newDPK.getKey().length > 0);
+
+        assertFalse("newKey should not be the same as the original deleted key", Arrays.equals
+                (dpk.getKey(), newDPK.getKey()));
```

Comment:
Given the keys are randomly generated this isn't strictly true. However, my guess is that duplications should be so rare we need not worry.